packages/block-library/src/navigation/use-is-collapsed.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { debounce, useMediaQuery } from '@wordpress/compose';
+import { useState, useLayoutEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { NAVIGATION_MOBILE_COLLAPSE } from './constants';
+import navigationIsWrapping from './is-wrapping';
+
+function useIsCollapsed( overlayMenu, navRef ) {
+	const isMobileBreakPoint = useMediaQuery(
+		`(max-width: ${ NAVIGATION_MOBILE_COLLAPSE })`
+	);
+
+	const shouldBeCollapsed = () => {
+		return (
+			( 'mobile' === overlayMenu && isMobileBreakPoint ) ||
+			'always' === overlayMenu ||
+			( 'auto' === overlayMenu && navigationIsWrapping( navRef.current ) )
+		);
+	};
+
+	const [ isCollapsed, setIsCollapsed ] = useState( shouldBeCollapsed() );
+
+	useLayoutEffect( () => {
+		function updateIsCollapsed() {
+			setIsCollapsed( shouldBeCollapsed() );
+		}
+		window.addEventListener( 'resize', debounce( updateIsCollapsed, 100 ) );
+		return () => {
+			window.removeEventListener( 'resize', updateIsCollapsed );
```

Comment:
Given that the function attached to the resize event is debounced, I'm not sure this will work.

Reply:
Correct. You'll have to define a variable that refers to the debounced function and then provide _that_ variable to the `removeEventListener`. Debounce returns a new function reference.